src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -106,41 +141,102 @@ private async Task<T> GetSystemDocumentByIdAsync<T>(string id, string partitionK
 
             using (cosmosDocumentQuery)
             {
-                FeedResponse<dynamic> response = await cosmosDocumentQuery.ExecuteNextAsync(cancellationToken);
-                var result = response.SingleOrDefault();
+                FeedResponse<Document> response = await cosmosDocumentQuery.ExecuteNextAsync<Document>(cancellationToken);
+                return response;
+            }
+        }
+
+        private async Task<T> GetSystemDocumentByIdAsync<T>(string id, string partitionKey, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(id, KnownDocumentProperties.Id);
+            var kvps = new List<KeyValuePair<string, object>> { new KeyValuePair<string, object>(KnownDocumentProperties.Id, id) };
```

Comment:
let's not do this. If it's retrieving document by id, it's much cheaper (in terms of RU) to write specific query for retriving docuemnt by id rather than a query.